[DOCS] Adds user-facing docs for the new KP logging configuration (#94993)
elastic/kibana: docs/migration/migrate_8_0.asciidoc

[float]
==== Default logging timezone is now the system's timezone
*Details:* In prior releases the timezone used in logs defaulted to UTC. We now use the host machine's timezone by default.

> *Impact:* To restore the previous behavior, in kibana.yml use the pattern layout, with a <<date-format, date modifier>>:

Reviewer: These breaking changes are also published in the Installation and Upgrade Guide (https://www.elastic.co/guide/en/elastic-stack/master/kibana-breaking-changes.html), so you can't use internal links like this (unless the target *also* exists in the Installation and Upgrade Guide). Instead, use an external link:

```suggestion
*Impact:* To restore the previous behavior, in kibana.yml use the pattern layout, with a {kibana-ref}/date-format.html[date modifier]:
```

Author: Ah, thank you!

Author: @lcawl It's going to be easier to add these links once the main logging documentation is merged. Thanks for the guidance!

[source,yaml]
-------------------
logging:
  appenders:
    console:
      type: console
      layout:
        type: pattern
        pattern: "%date{ISO8601_TZ}{UTC}"
-------------------
See https://github.com/elastic/kibana/pull/90368 for more details.

[float]
==== Responses are never logged by default

> *Details:* Previously responses would be logged if either `logging.json` was true, `logging.dest` was specified, or a `TTY` was detected. With the new logging configuration, these are provided by a dedicated logger.

Reviewer 2: > With the new logging configuration,

Should we provide a link to the logging service config? Ideally, in all the places when we refer to it.

Author: That is planned for the follow up PR. We can't seem to limbo to docs we're adding from the same PR.

> *Details:* Previously responses would be logged if either `logging.json` was true, `logging.dest` was specified, or a `TTY` was detected. With the new logging configuration, these are provided by a dedicated <<dedicated-loggers,logger>>.

Reviewer: Likewise here, you need to either make this an external link or else exclude it from the "notable-breaking-changes" section of this page:

```suggestion
*Details:* Previously responses would be logged if either `logging.json` was true, `logging.dest` was specified, or a `TTY` was detected. With the new logging configuration, these are provided by a dedicated {kibana-ref}/logging-service.html#dedicated-loggers[logger].
```

Author: See https://github.com/elastic/kibana/pull/94993#discussion_r597338769

> *Impact:* To restore the previous behavior, in `kibana.yml` enable `debug` for the <<request-response-logger,`http.server.response`>> logger:

Reviewer: Ditto re content re-used outside the Kibana book:
```suggestion
*Impact:* To restore the previous behavior, in `kibana.yml` enable `debug` for the {kibana-ref}/logging-service.html#request-response-logger[`http.server.response`] logger:
```

Author: See https://github.com/elastic/kibana/pull/94993#discussion_r597338769

[source,yaml]
-------------------
logging:
  loggers:
    - name: http.server.response
      appenders: [console]
      level: debug
-------------------
See https://github.com/elastic/kibana/pull/87939 for more details.

[float]
==== Logging destination is specified by the appender

> *Details:* Previously log destination would be `stdout` and could be changed to `file` using `logging.dest`. With the new logging configuration, you can specify the destination using <<logging-appenders,appenders>>. 

Reviewer: Ditto:
```suggestion
*Details:* Previously log destination would be `stdout` and could be changed to `file` using `logging.dest`. With the new logging configuration, you can specify the destination using {kibana-ref}/logging-service.html#logging-appenders[appenders]. 
```

Author: See https://github.com/elastic/kibana/pull/94993#discussion_r597338769

> *Impact:* To restore the previous behavior and log records to *stdout*, in `kibana.yml` use the `console` appender.

Reviewer 2: use an appender with `type: console`?

Author: You are right. I'll update accordingly.

-------------------

[float]
==== Set log verbosity with root

> *Details:* Previously logging output would be specified by `logging.silent` (none), `logging.quiet` (error messages only) and `logging.verbose` (all). With the new logging configuration, set the minimum required <<log-level,log level>>.

Reviewer: ```suggestion
*Details:* Previously logging output would be specified by `logging.silent` (none), `logging.quiet` (error messages only) and `logging.verbose` (all). With the new logging configuration, set the minimum required {kibana-ref}/logging-service.html#log-level[log level].
```

Author: See https://github.com/elastic/kibana/pull/94993#discussion_r597338769

-------------------

[float]
==== Declare log message format

> *Details:* Previously all events would be logged in `json` format when `logging.json` was true. With the new logging configuration you can specify the output format with <<logging-layouts,layouts>>. You can choose between <<logging-json-format-old-and-new-example,`json`>> and <<logging-pattern-format-old-and-new-example,pattern>> format depending on your needs.

Reviewer: ```suggestion
*Details:* Previously all events would be logged in `json` format when `logging.json` was true. With the new logging configuration you can specify the output format with {kibana-ref}/logging-service.html#logging-layouts[layouts]. You can choose between {kibana-ref}//logging-configuration-migration.html#logging-json-format-old-and-new-example[`json`] and {kibana-ref}/logging-configuration-migration.html#logging-pattern-format-old-and-new-example[pattern] format depending on your needs.
```

> *Impact:* To restore the previous behavior, in `kibana.yml` configure the logging format for each custom appender with the <<logging-layouts,`appender.layout`>> property. There is no default for custom appenders and each one must be configured expilictly. 

Reviewer: ```suggestion
*Impact:* To restore the previous behavior, in `kibana.yml` configure the logging format for each custom appender with the {kibana-ref}/logging-service.html#logging-layouts[`appender.layout`] property. There is no default for custom appenders and each one must be configured explicitly. 
```

==== Configure log rotation with the rolling-file appender
*Details:* Previously log rotation would be enabled when `logging.rotate.enabled` was true.

> *Impact:* To restore the previous behavior, in `kibana.yml` use the <<rolling-file-appender,`rolling-file`>> appender.

Reviewer: ```suggestion
*Impact:* To restore the previous behavior, in `kibana.yml` use the {kibana-ref}/logging-service.html#rolling-file-appender[`rolling-file`] appender.
```
